
record Error {
    msg : String;
}

instance Exception = Error;

error(...e) {
    var buf = String();
    printlnTo(buf, ...e);
    throw Error(move(buf));
}

errorNoThrow(...e) {
    printlnTo(stderr, "error: ", ...e);
}

inlined assert(cond:Bool) {
    if (not cond)
        error("assertion failed");
}

inlined overload assert(cond:Bool, message) {
    if (not cond)
        error("assertion failed: ", message);
}
